File: src/main/java/com/xgarage/app/model/Car.java
package com.xgarage.app.model;

import lombok.*;

import jakarta.persistence.*;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Car {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long brandId;

    @Column(nullable = false)
    private Long carModelId;

    @Column(nullable = false)
    private Long carModelYearId;

    @Column(nullable = false)
    private Long carModelTypeId;

    @Column(nullable = false)
    private String chassisNumber;

    @OneToOne
    @JoinColumn(name = "document_id", referencedColumnName = "id")
    private Document document;

    private String plateNumber;

    @Enumerated(EnumType.STRING)
    private GearType gearType;
}

File: src/app/xgarage/core/model/car.ts
import { GenericEntity } from "../../common/generic/genericentity";
import { Brand } from "../../common/model/brand";
import { CarModel } from "../../common/model/carmodel";
import { CarModelType } from "../../common/model/carmodeltype";
import { CarModelYear } from "../../common/model/carmodelyear";
import { GearType } from "./geartype";

export interface Car extends GenericEntity{
    brandId?: Brand;
    carModelId?: CarModel;
    carModelYearId?: CarModelYear;
    carModelTypeId?: CarModelType;
    chassisNumber?: string;
    plateNumber?: string;
    gearType?: GearType;
    document?: {extension: string, id: number, name:string};
}
